mongo/integration/unified/unified_spec_runner.go

```diff
@@ -151,8 +154,8 @@ func (tc *TestCase) EndLoop() {
 	tc.loopDone <- struct{}{}
 }
 
-// Run runs the TestCase
-func (tc *TestCase) Run(mt *mtest.T) {
+// Run runs the TestCase and returns an error if it fails
+func (tc *TestCase) Run(mt *mtest.T) error {
```

Comment:
I'd prefer for `Run` to not take a full `mtest.T` parameter because a future author could unknowingly call `Fatal` inside the function rather than returning an error. We do need a subset of the functionality in this function to do skipping/logging, though, so you can define an interface for those pieces and take that instead.

Reply:
done